CHANGELOG.md

```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 - Renamed `dispatcher` service to `eth-input-reader`
+- Refactored how `state-fold` is used inside `eth-input-reader`
```

Comment:
The main goal of the changelog is for the users to understand the changes so they can adapt their code. In our case, the main user is sunodo.

Here are some useful entries to the changelog:

1. Removed state-fold service
2. Renamed dispatcher service to eth-input-reader
3. Remove config variables X, Y, Z
4. Changed the name of config variables from A to B

We shouldn't mention internal changes that do not affect the user. E.g., refactors, internal modifications, etc.

Reply:
> The main goal of the changelog is for the users to understand the changes so they can adapt their code

I agree that's the main goal, but not the only one. As a contributor/developer, I find some refactors and internal modifications relevant enough to also be in the changelog.

I don't trust my memory, so if later on we find a bug in `eth-input-reader`, it would be helpful to look at the changelog and see that we ditched the `state-server` in this version and that might be a good starting point for troubleshooting.

I guess the real discussion is what is a "[notable change](https://keepachangelog.com/en/1.0.0/#what)". IMO this is notable enough.

Reply:
Yes, I agree, but it should be more descriptive, like you said. We can add multiple entries in the changelog as well.